src/app/model/ChessModel.ts

```diff
@@ -1,9 +1,11 @@
 import { Square, SquareWithPiece } from './Types';
 import { PieceWasMoved } from './PieceWasMoved';
 import { PieceWasCaptured } from './PieceWasCaptured';
+import { KingWasChecked } from './KingWasChecked';
+import { KingWasUnchecked } from './KingWasUnchecked';
 
 export interface ChessModel {
   readonly squaresWithPiece: SquareWithPiece;
   possibleMoves(position: Square): Square[];
-  move(squareFrom: Square, squareTo: Square): (PieceWasMoved | PieceWasCaptured)[];
+  move(squareFrom: Square, squareTo: Square): (PieceWasMoved | PieceWasCaptured | KingWasChecked | KingWasUnchecked)[];
```

Comment:
Teraz tutaj też możesz wprowadzić ten typ. To w sumie `MoveResult` moze powędrować do tego pliku albo do osobnego - dedykowanego :)

Reply:
Bump!